src/Eccube/Repository/PageLayoutRepository.php

```diff
@@ -138,6 +138,10 @@ public function getByUrl(DeviceType $DeviceType, $url)
             ))
             ->getSingleResult();
 
+        if(count($ownResult->getBlockPositions()) && ($url == 'preview') && ($page == 'homepage')) {
```

Comment:
I don't understand.
Please teach me the reason you are using `($page == 'homepage')`.